### PR TITLE
fix: remove discrim validation for updating username

### PR DESCRIPTION
**Pull request by zt64**
Removed the validation for discriminator, and also hide's the discriminator input entirely if the user has already been moved to the new username system.

### CODE REVIEW

File: Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt [hunk 1]
```diff
@@ -150,6 +158,27 @@ fun patchDefaultAvatars() {
     )
 }
 
+fun patchUsername() {
+    Patcher.addPatch(
+        `AuthUtils$createDiscriminatorInputValidator$1`::class.java.getDeclaredMethod("getErrorMessage", TextInputLayout::class.java),
+        InsteadHook.DO_NOTHING
```
Comment (rushiiMachine) [inline Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:164]: What about those still with a legacy discrim + nitro? Or can you not even change discrim at all anymore

Comment (Kyiro) [reply Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:164]: Yeah you can't update your discrim anymore, it'll only get updated when your nitro runs out.

Comment (dolfies) [reply Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:164]: Not true, you can still change it with Nitro

Comment (rushiiMachine) [reply Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:164]: in that case this check shouldn't be disabled if the user isn't pomelo'd yet

File: Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt [hunk 2]
```diff
@@ -4,9 +4,12 @@
  * Licensed under the Open Software License version 3.0
  */
 
+@file:Suppress("MISSING_DEPENDENCY_SUPERCLASS")
+
```
Comment (rushiiMachine) [inline Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:8]: can u remove

File: Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt [hunk 3]
```diff
@@ -150,6 +159,24 @@ fun patchDefaultAvatars() {
     )
 }
 
+fun patchUsername() {
+    if (StoreStream.getUsers().me.discriminator == 0) {
```
Comment (rushiiMachine) [inline Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:163]: you could use an early return here instead to reduce nesting

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

File: Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt (hunk 2 above)
Comment (zt64) [reply Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:8]: But it gets rid of the error for those funny classes
I'll remove if it's a must

File: Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt (hunk 3 above)
Comment (zt64) [reply Aliucord/src/main/java/com/aliucord/coreplugins/rn/Patches.kt:163]: Oh yea I'll do that

**Review by rushiiMachine — APPROVED** [review]

**Pull request merged**